### ISSUE TITLE
Build logs scrambled (and truncated) in 1.3.1

### ISSUE THREAD

**Issue by rfay** [surf-build/surf#73]
I tried updating surf-build to 1.3.1 (from 1.0.4) and noted that the functionality was fine, but the log was scrambled. For example, see https://gist.github.com/drud-test-machine-account/f6b75c2144abd4b6cffab755d60ddbd6 - it shows the completion message (from the end) first, then some random stuff from the middle of the log, then most of the log is truncated.

**Comment by mato** [comment]
@rfay:

Based on the docs for [fs.write()](https://nodejs.org/api/fs.html#fs_fs_write_fd_string_position_encoding_callback), the problem is here:

https://github.com/surf-build/surf/blob/a4ca24e6ec97a0f9c985b9955df24e6393db33ba/src/build-project-main.ts#L196

What happens is that each individual write just overwrites whatever is starting at offset `0` in the build log.

However: Changing the `0` to `null` gives me the following error when building:

```
src/build-project-main.ts(196,50): error TS2345: Argument of type 'string' is not assignable to parameter of type 'Buffer'.

196       return Observable.fromPromise(fs.write(fd, x, null, 'utf8'));
                                                     ~
```

Unfortunately my Node/Typescript-fu is not up to fixing that error.

Also, per the docs mentioned above, shouldn't this be using [fs.createWriteStream()](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options) instead?

**Comment by mato** [comment]
I can confirm that manually patching the compiled `build-project-main.js` with the above fixes the problem. I've no idea how to fix the Typescript stuff.

**Comment by paulcbetts** [comment]
Fixed in 5b4af2268eb23ed4492a5ff0fd4440eb003a2649, sorry about that all
